app/assets/javascripts/pretty-text/engines/discourse-markdown-it.js.es6

```diff
@@ -124,64 +124,73 @@ function setupHoister(md) {
   md.renderer.rules.html_raw = renderHoisted;
 }
 
+function extractDataAttribute(str) {
+  let sep = str.indexOf("=");
+  if (sep === -1) {
+    return null;
+  }
+
+  const key = `data-${str.substr(0, sep)}`.toLowerCase();
+  if (!/^[A-Za-z]+[\w\-\:\.]*$/.test(key)) {
+    return null;
+  }
+
+  const value = str.substr(sep + 1);
+  return [key, value];
+}
+
 const IMG_SIZE_REGEX = /^([1-9]+[0-9]*)x([1-9]+[0-9]*)(\s*,\s*(x?)([1-9][0-9]{0,2}?)([%x]?))?$/;
 function renderImage(tokens, idx, options, env, slf) {
-  var token = tokens[idx];
-
-  let alt = slf.renderInlineAsText(token.children, options, env);
-
-  let split = alt.split("|");
-  if (split.length > 1) {
-    let match;
-    let info = split.splice(split.length - 1)[0];
-
-    if ((match = info.match(IMG_SIZE_REGEX))) {
-      if (match[1] && match[2]) {
-        alt = split.join("|");
-
-        let width = match[1];
-        let height = match[2];
-
-        // calculate using percentage
-        if (match[5] && match[6] && match[6] === "%") {
-          let percent = parseFloat(match[5]) / 100.0;
-          width = parseInt(width * percent, 10);
-          height = parseInt(height * percent, 10);
-        }
-
-        // calculate using only given width
-        if (match[5] && match[6] && match[6] === "x") {
-          let wr = parseFloat(match[5]) / width;
-          width = parseInt(match[5], 10);
-          height = parseInt(height * wr, 10);
-        }
-
-        // calculate using only given height
-        if (match[5] && match[4] && match[4] === "x" && !match[6]) {
-          let hr = parseFloat(match[5]) / height;
-          height = parseInt(match[5], 10);
-          width = parseInt(width * hr, 10);
-        }
-
-        if (token.attrIndex("width") === -1) {
-          token.attrs.push(["width", width]);
-        }
-
-        if (token.attrIndex("height") === -1) {
-          token.attrs.push(["height", height]);
-        }
-
-        if (
-          options.discourse.previewing &&
-          match[6] !== "x" &&
-          match[4] !== "x"
-        )
-          token.attrs.push(["class", "resizable"]);
+  const token = tokens[idx];
+  const alt = slf.renderInlineAsText(token.children, options, env);
+
+  const split = alt.split("|");
+  const alt_split = [];
+
+  for (let i = 0, match, data; i < split.length; ++i) {
```

Comment:
I think it could be interesting to have variables for each used match. eg: `const foo = match[5]`

This would make the whole code block much more readable.

Reply:
The problem is that some parts may have multiple meanings. An "accurate" name for `match[5]` would be `widthOrHeightRatioOrPercentage`, which is useless. Later, variables are defined to hold the value (for example for `match[5]` it is `percent`, `wr` or `hr`).